### PR TITLE
Call transfer.withUserOp trpc mutation from frontend

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/688?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#688** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/688?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#687** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/687?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#686** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/686?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/app/utils/useSignTransferUserOpMutation.ts
Comment (0xBigBoss) [inline packages/app/utils/useSignTransferUserOpMutation.ts:1]: mistake?

File: packages/app/features/send/confirm/screen.tsx [hunk 1]
```diff
@@ -175,10 +180,37 @@ export function SendConfirm() {
...
+      const signature = await signUserOp({
+        userOpHash,
+        allowedCredentials:
+          webauthnCreds?.map((c) => ({
+            id: byteaToBase64(c.raw_credential_id),
+            userHandle: c.name,
+          })) ?? [],
       })
+
+      const receipt = await transfer({
```
Comment (0xBigBoss) [inline packages/app/features/send/confirm/screen.tsx:199]: I don't think you're returning the receipt anymore, but we should return the workflow id or something so the UI can start querying the status

**Review by 0xBigBoss — COMMENTED** [review]
Amazing work sir. This is a step in the right direction. We can continue to improve along the way.

There's a ton of nice to have improvements but we potentially don't need all of them.

1. Workflow Structure:
   The `TransferWorkflow` is well-structured with clear, sequential steps, making it easy to understand and maintain. Great job breaking it down into distinct activities!

2. Error Handling:
   You've implemented good error handling with `ApplicationFailure.nonRetryable` and `ApplicationFailure.retryable`. This distinction is crucial in Temporal workflows. Consider adding more detailed error messages to help with debugging.

3. Activity Timeouts:
   The current timeout of 30 seconds for all activities might not be sufficient for network operations. Consider adjusting these based on the expected duration of each activity, especially for blockchain interactions.

4. Workflow Input Validation:
   While you're validating inputs in the `sendUserOpActivity`, consider moving some of this validation to the workflow level to fail fast if invalid data is provided.

5. Observability and User Feedback:
   To improve observability and provide valuable feedback to users, let's focus on the following: [see temporal example](https://github.com/temporalio/samples-typescript/tree/692fd263925a32add2c14fe7d7f2edc28b71b4ce/timer-progress)

   a. Add statuses to the activities such as `sending`, `waiting for receipt`, `confirming`.
   b. Implement a tRPC route for getting the workflow status. This can return the activity status so the user has feedback on what's happening.
   c. Add more logging throughout the workflow and activities using Temporal's built-in logging.

   Here's a basic example of how you might implement status tracking and querying:

   ```typescript
   // In your workflow file
   export const getStatusQuery = defineQuery<string>('getStatus');

   export async function TransferWorkflow(args: TransferWorkflowArgs) {
     let status = 'initialized';
     setHandler(getStatusQuery, () => status);

     status = 'sending';
     const hash = await sendUserOpActivity(args);
     // ... rest of the workflow
   }

   // In your tRPC route
   export const transferRouter = createTRPCRouter({
     // ... other routes
     getStatus: protectedProcedure
       .input(z.string())
       .query(async ({ input: workflowId }) => {
         const handle = client.workflow.getHandle(workflowId);
         return handle.query(getStatusQuery);
       }),
   });
  ```

**Comment by youngkidwarrior** [comment]
Nice! Yeah, this isn't done haha.  Not returning the receipt or anything.

Just didn't want to keep building it out if there are any big flaws I wasn't noticing

File: packages/app/utils/useUserOpTransferMutation.test.ts [hunk 2]
```diff
@@ -1,331 +1,331 @@
...
-    )
-
-    await waitFor(() => expect(result.current.isError).toBe(true))
-
-    expect(result.current).toBeDefined()
-    expect(result.current.error).toBeDefined()
-    expect(result.current.error?.message).toBe('Invalid nonce')
-  })
-})
+// import { describe, test } from '@jest/globals'
```
[the remark this answers is not in the document]
Comment (youngkidwarrior) [reply packages/app/utils/useUserOpTransferMutation.test.ts:1]: I didn't mean to do this lol

**Pull request closed without merging**